islandora_pathauto.module

```diff
@@ -319,3 +322,95 @@ class IslandoraFedoraObjectReadOnly implements ArrayAccess {
    */
   public function offsetUnset($offset) {}
 }
+
+/*
+ * Implements hook_form_alter()
+ * 
+ * Adds a TextField to Islandora Properties so the User can Change
+ * the URL Alias.
+ */
+function islandora_pathauto_form_alter(&$form , &$form_state, $form_id){
+    if( $form_id == 'islandora_object_properties_form' ){
+
+        if( !pathauto_pattern_load_by_entity('islandora') ){
+            $hidden = 'hidden';
+        }
+        else{
+            $hidden = 'checkbox';
+        }
+                
+        $generate_url = variable_get( 'set_generate', 0 );
```

Comment:
`variable_get()` and `variable_set()` names should be module specific.  Note also that these functions set sitewide variables, so if one user looked at an object which was autogenerating and another user looked at an object which was not autogenerating, you would have a conflict.  I think you want a hidden form field here instead, representing whether the current alias matches what would be generated by pathauto.